### PR TITLE
userId can get and specify create parameter

### PR DESCRIPTION
**Pull request by shoota**
# TL;DR

This PR resolve to sharing userId on cross domain tracking used iframe.

- userId should be able to get through global interface.
- tracker can create with userId

```
_ud("get", "clientId") // => return userId string
_ud("create", { clientId: client_id_from_tracker })
```

### How to check

*   checkout this pr.
*   yarn install && yarn build
*  yarn test

### Check list

*   [x] pass the tests

### CODE REVIEW

File: packages/agent/src/agent.ts [hunk 1]
```diff
@@ -89,10 +89,10 @@ export default class Agent {
     return this.core.mergeDeep(key)
   }
 
-  public get (key: string): string {
+  public get (key: StateKey | 'linkerParam'): ClientEnvironmentsData | CustomData | string {
     return key === 'linkerParam'
       ? `${this.linkerName}=${this.core.get('userId')}`
-      : ''
+      : this.core.get(key)
```
Comment (shoota) [inline packages/agent/src/agent.ts]: get can return core store data

File: packages/agent/src/core.ts [hunk 2]
```diff
@@ -91,7 +91,7 @@ export default class AgentCore extends Store {
     eventsClass: any[], // TODO
     settings: SettingFieldsObject
   ) {
-    const userId = findOrCreateUserId(settings, getLocation())
+    const userId = settings.userId ? settings.userId : findOrCreateUserId(settings, getLocation())
```
[the remark this answers is not in the document]
Comment (shoota) [reply packages/agent/src/core.ts]: need userId validation?

[the remark this answers is not in the document]
Comment (heavenshell) [reply packages/agent/src/core.ts]: Tracker Server should validate userId
So, agent don't need validation.

[the remark this answers is not in the document]
Comment (Surgo) [reply packages/agent/src/core.ts]: 🙆

**Pull request merged**